plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/model/CodeTransformTelemetryMetadataSingleton.kt

```diff
@@ -0,0 +1,18 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.codemodernizer.model
+
+object CodeTransformTelemetryMetadataSingleton {
```

Comment:
i like the modularization for testing, but now your project service is sharing state across all instances, which is probably not what you want

Reply:
We only have one instance at a time right now and we do want to share this across any reference to it. For simplicity we are not managing the state of the object within any of our manager classes.

If we move to a multi instance service, we can make this a variable with session or sessionContext.

Reply:
in the interest of avoiding global state that isn't managed by the platform then, can you convert this into a light service?
https://plugins.jetbrains.com/docs/intellij/plugin-services.html#light-services

your test will need to have an `ApplicationRule` applied

Reply:
Sure. I'll check out adding these changes to convert to:

```
@Service(Service.Level.PROJECT)
public final class MyProjectService {
...
```